### PR TITLE
align project directory structure

### PR DESCRIPTION
**Pull request by fgimenez**
Towards https://github.com/giantswarm/giantswarm/issues/1666

In this case the spec is simple, no need for a package, cleared the `spec.go` file leaving only the `Spec` struct in it.

There are additional functionalities and data structures in the TPR not strictly related to the resource, not sure if they really belong here, or if I should move them into a specific package here, let's discuss in the review.

### CODE REVIEW

File: spec.go [hunk 1]
```diff
@@ -1,9 +1,5 @@
 package certificatetpr
 
-type Searcher interface {
-	SearchCerts(clusterID string) (AssetsBundle, error)
-}
-
```
Comment (xh3b4sd) [inline spec.go]: Why does this change? IMO it was correct before.

**Review by xh3b4sd — CHANGES_REQUESTED** [review]

**Comment by rossf7** [comment]
For the other TPRs there are tests for the YAML encoding. I think these are useful to prevent us making breaking changes to the structure. See https://github.com/giantswarm/ingresstpr/blob/master/spec_test.go

So I think there should be a test for this.

File: spec.go (hunk 1 above)
Comment (fgimenez) [reply spec.go]: Reverted, thx @xh3b4sd!

**Comment by fgimenez** [comment]
Good point @rossf7, we already have https://github.com/giantswarm/certificatetpr/blob/master/spec_test.go, do you think it should be extended?

**Review by rossf7 — APPROVED** [review]
Ah OK, I missed the test was already there! I don't think it needs to be extended. So this LGTM.

On the extra functionality, the searching for the secrets is needed by both AWS and KVM operators. IMO it's OK here and doesn't need to be extracted to a separate repo.

**Review by xh3b4sd — APPROVED** [review]

**Comment by kopiczko** [comment]
Can you show output of `tree -I vendor* ./`?

**Comment by fgimenez** [comment]
@kopiczko sure:
```
$ tree -I vendor* ./
./
├── certificatetprtest
│   ├── service.go
│   └── service_test.go
├── circle.yml
├── custom_object.go
├── error.go
├── glide.lock
├── glide.yaml
├── LICENSE
├── list.go
├── meta.go
├── README.md
├── service.go
├── spec.go
├── spec_test.go
├── testdata
│   └── spec.yaml
├── tls_assets.go
└── tls_assets_test.go

2 directories, 17 files
```

**Review by kopiczko — APPROVED** [review]
LGTM. I closed by mistake. Sorry :)

**Comment by fgimenez** [comment]
np :)

**Pull request merged**